lib/services/cloud-projects-service.ts

```diff
@@ -0,0 +1,53 @@
+import { ExportOptions } from "../constants";
+
+export class CloudProjectsService implements ICloudProjectsService {
+	constructor(private $remoteProjectService: IRemoteProjectService,
+		private $errors: IErrors,
+		private $prompter: IPrompter) { }
+
+	public async getSolutionProjectInfo(opts: ISolutionProjectPairOptions): Promise<ISolutionProjectInfo> {
+		const apps = await this.$remoteProjectService.getAvailableAppsAndSolutions();
+		let framework = "";
+		let id = "";
+		const noSolutionNamePassed = !opts.solutionName;
+
+		if (noSolutionNamePassed) {
+			const solutionNames = apps.map(sln => sln.colorizedDisplayName) || [];
+			opts.solutionName = await this.$prompter.promptForChoice("Select solution to export", solutionNames);
```

Comment:
what if the terminal is not interactive?

Reply:
this is handled in the command's `canExecute` method